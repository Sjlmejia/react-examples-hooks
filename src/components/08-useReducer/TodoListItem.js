import React from 'react'

export const TodoListItem = ({description, id, done, index, handleToggle, handleDelete}) => {
  return (
    <li key={id} className="list-group-item">
      <div
        className={`item ${done && 'complete'}`}
        onClick={() => { handleToggle(id)}}
      >
        {index +1}. {description}
      </div>
      <button className="btn btn-outline-danger" onClick={()=> {handleDelete(id)}}>Borrar</button>
    </li>
  )
}
