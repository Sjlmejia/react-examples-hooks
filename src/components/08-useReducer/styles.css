body {
  padding: 50px;
}
.list-group-item {
  display: flex;
  justify-content: space-between;
}

.item {
  cursor: pointer;
  display: flex;
  align-items: center;
}

.complete {
  text-decoration: line-through;
}