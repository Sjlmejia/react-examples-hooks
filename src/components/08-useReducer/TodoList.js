import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({todos,handleDelete, handleToggle }) => {
  return (
    <ul className="list-group list-group-flush">
        {
          todos.map(({desc, id, done}, i) => {
            return <TodoListItem
                      key={id}
                      description={desc}
                      id={id} done={done}
                      index={i}
                      handleToggle={handleToggle}
                      handleDelete={handleDelete}/>
          })
        }
    </ul>
  )
}
