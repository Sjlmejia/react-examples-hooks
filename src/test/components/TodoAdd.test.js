import React from 'react';
import { shallow } from 'enzyme';
import {TodoAdd} from '../../components/08-useReducer/TodoAdd'

describe('prueba es todo Add', () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd  handleAddTodo={ handleAddTodo}/>)

  test('deberia montarse el comopente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('no debe llamar el handleaddTodo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({preventDefault(){} });
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test('debe de llamar la funcion HandleaddTodo', () => {
    const value = 'Aprender React';
    wrapper.find('input').simulate('change', {
      target:{
        value,
        name: 'description'
      }
    });
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({preventDefault(){} });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith({
      desc: value,
      done: false,
      id:expect.any(Number)
    });
    expect( wrapper.find('input').prop('value')).toBe('');
  });
  
  
});
