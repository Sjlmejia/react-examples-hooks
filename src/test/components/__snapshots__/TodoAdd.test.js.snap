// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba es todo Add deberia montarse el comopente 1`] = `
<div>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1"
    >
      Agregar
    </button>
  </form>
</div>
`;
